### ISSUE TITLE
Caching is not a valid parameter for REST API

### ISSUE THREAD

**Issue by pawelpabich** [hdinsight/hbase-sdk-for-net#35]
More details here: https://hbase.apache.org/apidocs/org/apache/hadoop/hbase/rest/package-summary.html#operation_cell_query_multiple. It looks like caching is only available in the JAVA RPC API.

**Comment by thomasjungblut** [comment]
Pawel, I would strongly recommend you to not alter generated protobuf files.
If it is only available in Java RPC, we should open a ticket with HBase and fix it there.

**Comment by pawelpabich** [comment]
Sure, but for now this will remove confusion.  We've spent hours trying to figure out why this does not work.

**Comment by duoxu** [comment]
@pawelpabich I tried this yesterday and it works.

batch is the number of column values returned per request.
cache is the number of rows prefetched from regionserver to REST server, I believe.

Larger batch size will reduce round trips between client and REST server.
Larger cache size will reduce round trips between REST server and regionserver.

I did an experiment, I tried to scan a table of 10000 rows with 3 columns each row.

batch size was set to 10000, so a single getNext request should return all the values.
I tried cache size = 1 vs 10000, and result is 5sec vs 1sec.

Definitely the large cache size was helpful in this case.
